core/app/repo/common.go

```diff
@@ -28,57 +30,56 @@ func NewICommonRepo() ICommonRepo {
 	return &CommonRepo{}
 }
 
-func (c *CommonRepo) WithByID(id uint) DBOption {
+func (c *CommonRepo) WithByID(id uint) global.DBOption {
 	return func(g *gorm.DB) *gorm.DB {
 		return g.Where("id = ?", id)
 	}
 }
-func (c *CommonRepo) WithByIDs(ids []uint) DBOption {
+func (c *CommonRepo) WithByGroupID(id uint) global.DBOption {
 	return func(g *gorm.DB) *gorm.DB {
-		return g.Where("id in (?)", ids)
+		return g.Where("group_id = ?", id)
 	}
 }
-func (c *CommonRepo) WithByName(name string) DBOption {
+
+func (c *CommonRepo) WithByIDs(ids []uint) global.DBOption {
 	return func(g *gorm.DB) *gorm.DB {
-		if len(name) == 0 {
-			return g
-		}
-		return g.Where("`name` = ?", name)
+		return g.Where("id in (?)", ids)
 	}
 }
-func (c *CommonRepo) WithLikeName(name string) DBOption {
+func (c *CommonRepo) WithByName(name string) global.DBOption {
 	return func(g *gorm.DB) *gorm.DB {
-		if len(name) == 0 {
-			return g
-		}
-		return g.Where("name like ? or command like ?", "%"+name+"%", "%"+name+"%")
+		return g.Where("`name` = ?", name)
 	}
 }
-func (c *CommonRepo) WithByType(ty string) DBOption {
+
+func (c *CommonRepo) WithByType(ty string) global.DBOption {
 	return func(g *gorm.DB) *gorm.DB {
-		if len(ty) == 0 {
-			return g
-		}
 		return g.Where("`type` = ?", ty)
 	}
 }
-func (c *CommonRepo) WithByKey(key string) DBOption {
+func (c *CommonRepo) WithByKey(key string) global.DBOption {
 	return func(g *gorm.DB) *gorm.DB {
 		return g.Where("key = ?", key)
 	}
 }
-func (c *CommonRepo) WithByStatus(status string) DBOption {
+func (c *CommonRepo) WithByStatus(status string) global.DBOption {
 	return func(g *gorm.DB) *gorm.DB {
 		return g.Where("status = ?", status)
 	}
 }
-func (c *CommonRepo) WithOrderBy(orderStr string) DBOption {
+func (c *CommonRepo) WithByGroupBelong(group string) global.DBOption {
+	return func(g *gorm.DB) *gorm.DB {
+		return g.Where("group_belong = ?", group)
+	}
+}
+
+func (c *CommonRepo) WithOrderBy(orderStr string) global.DBOption {
 	return func(g *gorm.DB) *gorm.DB {
 		return g.Order(orderStr)
 	}
 }
 
-func (c *CommonRepo) WithOrderRuleBy(orderBy, order string) DBOption {
+func (c *CommonRepo) WithOrderRuleBy(orderBy, order string) global.DBOption {
 	switch order {
 	case constant.OrderDesc:
 		order = "desc"
```

Comment:
The provided code is incomplete and contains some errors. Here are my observations:

There seems to be an inconsistent usage of global.DBOption in multiple places, which needs to be fixed.

As for specific issues or optimizations, it's important to review each implementation based on real-world requirements.

However, based solely on the provided comments, I can make general notes that could help improve efficiency:
- The `goroutine` syntax should not used due to its deprecated nature.
- There isn't enough context available about how you're using goroutines inside this code snippet, but if used appropriately, they may introduce significant overheads in certain scenarios.
- Consider using Gorm functions more effectively instead of creating custom ones. For example, use `where` with SQL queries directly rather than manually building them using where clauses from database objects.

If there was supposed to be further information here such as additional interfaces, structures, method signatures or examples of data types, let me know!